SmartDeviceLink/SDLIAPSession.m

```diff
@@ -141,10 +258,26 @@ - (SDLStreamErrorHandler)streamErroredHandler {
     };
 }
 
+- (SDLStreamHasSpaceHandler)streamHasSpaceHandler {
+    __weak typeof(self) weakSelf = self;
+    
+    return ^(NSStream *stream) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        
+        if (strongSelf.isDataSession) {
+            NSError *sendErr = nil;
+            
+            if (![strongSelf sdl_dequeueAndWriteToOutputStream:&sendErr] && sendErr != nil) {
+                [strongSelf sdl_handleOutputStreamWriteError:sendErr];
+            }
+        }
+    };
+}
 
 #pragma mark - Lifecycle Destruction
 
 - (void)dealloc {
+    self.sendDataQueue = nil;
```

Comment:
I think the best way to proceed is as follows:
1. I will remove the deallocs from the PR

2. Let's discuss on Slack a process for ensuring we can reproduce the failures we are seeing with connection and disconnection transition cycles. When I joined the project, we had a number of open crash tickets in our JIRA due to disconnection while SDL communication was in flight, followed by reconnection shortly thereafter. I think it's best for us to roll up the ticket repro steps and provide them to you.

3. Let's agree on a path forward based on empirical metrics such as MTTF or mean cycles to failure that allows all parties to track robustness as connections come and go.